src/Umbraco.Web/PropertyEditors/NestedContentPropertyEditor.cs

```diff
@@ -184,9 +187,16 @@ public override object ToEditor(Property property, IDataTypeService dataTypeServ
                                 continue;
                             }
 
-                            var tempConfig = dataTypeService.GetDataType(prop.Value.PropertyType.DataTypeId).Configuration;
-                            var valEditor = propEditor.GetValueEditor(tempConfig);
-                            var convValue = valEditor.ToEditor(tempProp, dataTypeService);
+                            var dataTypeId = prop.Value.PropertyType.DataTypeId;
+                            if (!valEditors.ContainsKey(dataTypeId))
```

Comment:
valEditors.TryGetValue is a faster way to lookup and retrieve from a dictionary at the same time.